packages/ra-core/src/controller/useEditController.ts

```diff
@@ -55,16 +56,21 @@ export interface EditControllerProps {
  */
 const useEditController = (props: EditProps): EditControllerProps => {
     useCheckMinimumRequiredProps('Edit', ['basePath', 'resource'], props);
-    const { basePath, id, resource, undoable = true } = props;
+    const { basePath, id, resource, message, undoable = true } = props;
     const translate = useTranslate();
     const notify = useNotify();
     const redirect = useRedirect();
     const refresh = useRefresh();
     const version = useVersion();
     const { data: record, loading } = useGetOne(resource, id, {
         version, // used to force reload
-        onFailure: () => {
-            notify('ra.notification.item_doesnt_exist', 'warning');
+        onFailure: error => {
```

Comment:
I don't agree with this change. All around the application, if the `GET_ONE` call fails, we consider that it's a 404. If you change it here, it should be changed everywhere else. But I don't want to do that, so please revert it.